com.unity.render-pipelines.high-definition/Editor/Lighting/LightUnit/LightUnitSlider.cs

```diff
@@ -202,19 +240,141 @@ protected override float GetPositionOnSlider(float value)
         {
             value.floatValue = ExponentialSlider(rect, value.floatValue);
         }
 
         float ExponentialSlider(Rect rect, float value)
         {
-            var internalValue = GUI.HorizontalSlider(rect, ValueToSlider(value), 0f, 1f, GUI.skin.horizontalSlider, GUI.skin.horizontalSliderThumb);
-
+            var internalValue = GUI.HorizontalSlider(rect, ValueToSlider(value), 0f, 1f);
             return SliderToValue(internalValue);
         }
     }
 
+    /// <summary>
+    /// Formats the provided descriptor into a piece-wise linear slider with contextual slider markers, tooltips, and icons.
+    /// </summary>
+    class PiecewiseLightUnitSlider : LightUnitSlider
+    {
+        struct Piece
+        {
+            public Vector2 domain;
+            public Vector2 range;
+            public Func<float, float> transform;
+            public Func<float, float> inverseTransform;
+        }
+
+        // Piecewise function indexed by value ranges.
+        private readonly Dictionary<int, Piece> m_PiecewiseFunctionMap = new Dictionary<int, Piece>();
+
+        Func<float, float> GetTransformation(float x0, float x1, float y0, float y1)
+        {
+            var m = (y0 - y1) / (x0 - x1);
+            var b = (m * -x0) + y0;
+
+            return x => (m * x) + b;
+        }
+
+        // Ensure clamping to (0,1) as sometimes the function evaluates to slightly below 0 (breaking the handle).
+        float ValueToSlider(Piece piecewise, float x) => Mathf.Clamp01(piecewise.inverseTransform(x));
+        float SliderToValue(Piece piecewise, float x) => piecewise.transform(x);
+
+        // Ideally we want a continuous, monotonically increasing function, but this is useful as we can easily fit a
+        // distribution to a set of (huge) value ranges onto a slider.
+        public PiecewiseLightUnitSlider(LightUnitSliderUIDescriptor descriptor) : base(descriptor)
+        {
+            // Sort the ranges into ascending order
+            var sortedRanges = m_Descriptor.valueRanges.OrderBy(x => x.value.x).ToArray();
+            var sliderDistribution = m_Descriptor.sliderDistribution;
+
+            // Compute the transformation for each value range.
+            for (int i = 0; i < sortedRanges.Length; i++)
+            {
+                var r = sortedRanges[i].value;
+
+                var x0 = sliderDistribution[i + 0];
+                var x1 = sliderDistribution[i + 1];
+                var y0 = r.x;
+                var y1 = r.y;
+
+                Piece piece;
+                piece.domain = new Vector2(x0, x1);
+                piece.range  = new Vector2(y0, y1);
+
+                piece.transform = GetTransformation(x0, x1, y0, y1);
+
+                // Compute the inverse
+                CoreUtils.Swap(ref x0, ref y0);
+                CoreUtils.Swap(ref x1, ref y1);
+                piece.inverseTransform = GetTransformation(x0, x1, y0, y1);
```

Comment:
IMO swapping and then calling the method is a bit overkill here.

Just calling `GetTransformation(y0, y1, x0, x1)` is enough as the swapped values are not reused for something else.
I find it easier to understand the intent here.